proposals/pubsub.md

```diff
@@ -0,0 +1,182 @@
+- Ceramic uses pubsub for queries and log replication.
+- [OrbitDB][] depends on pubsub to do replication.
+- Fission team would like to use pubsub (has user requests, but found it unreliable)
+
+It also fragments network by (JS) nodes having it enabled and nodes that do not making IPNS unrelaible.
+
+> IPFS in browsers use IPNS over pubsub (no DHT for browsers) while other nodes do it over DHT.
+
+Enabling PubSub would populate network with more nodes making it more reliable.
+
+
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Teams building products need pubsub.
+- Disabled pubsub makes an ecosystem more complex.
+- Building (near) real-time application on IPFS requires pubsub.
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+They would just be able to use `pubsub` without `--enable-pubsub-experiment` flag. Feature becomes usable to ipfs-client's that use IPFS HTTP API.
+
+> **From Fission**:
+> 
+> If we expect apps to be independent and run on the IPFS network -- like they do with HTTP servers -- then whatever else can be made available natively can be powerful building blocks
+>
+> Having more capabilities available out of the box for developers is a better reason to build and host on IPFS natively. Pubsub out of the box gives a host of real time and messaging opportunities that our customers would love to use.
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+- PubSub becomes more reliable with more nodes on the network.
+- Bulding (near) real-time products becomes easier, IPFS nodes are suitable out of the box.
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+- Improves velocity of teams building products requiring pubsub by removing additional burden.
+- Improves pubsub reliability by adding more enabled nodes to the network.
+- Enables development of new features that require pubsub.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+Level 3
+
+- We know teams building products that depend pubsub.
+- Features like IPNS over pubsub depends on this.
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+- Measure what is the overhead on nodes that do not directly use pubsub
```

Comment:
Do we need a metric on "reliability" (or do we already have one)?  I'm assuming we'll want to look at this in the PL ipfs nodes and ensure "reliability" doesn't decrease.

Reply:
I would expect that denser mesh would improve reliability, but more qualified folks would know better. I would assume that these kind of measurements were done on test nets when feature was developed, but would be good to hear from someone who was involved.

@aschmahmann @Stebalien could you provide your feedback please.

Reply:
I assume this is responding to the line above https://github.com/protocol/web3-dev-team/pull/53/files/40e4cabcc8d312689f6e45a721bab036ca2bcecb#r583299853?